build.rs

```diff
@@ -59,4 +59,28 @@ fn main() {
             .file("src/cuda/ptx.cu")
             .compile("sloth256_189_cuda");
     }
+
+    if cfg!(feature = "opencl") {
+        env::var("DEP_OPENMP_FLAG").unwrap().split(" ").for_each(|f| { cc.flag(f); });
+
+        println!("cargo:rustc-link-lib=OpenCL");
+        println!("cargo:rustc-link-search=/opt/amdgpu-pro/lib64/");
+        println!("cargo:rustc-link-search=C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v11.4/lib/x64");
+        println!("cargo:rustc-link-search=C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v11.5/lib/x64");
```

Comment:
Can't these and some of the below be handled in a nicer way? I see there is `.cuda(true)` above, is it something different?

And for Intel/AMD as well unless not supported yet.

Reply:
This is for Windows support only. Adding a .cuda(true) and compiling the OpenCL files with nvcc may be possible, with a fallback to compilation without nvcc in case CUDA is not installed.